src/test/java/org/broadinstitute/hellbender/tools/spark/sv/KmerTest.java

```diff
@@ -0,0 +1,123 @@
+package org.broadinstitute.hellbender.tools.spark.sv;
+
+import org.broadinstitute.hellbender.utils.test.BaseTest;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+/**
+ * Created by tsharpe on 1/12/16.
+ */
+public class KmerTest extends BaseTest
+{
+    @Test
+    public void testDefaultConstruction()
+    {
+        Assert.assertEquals(new Kmer(10).toString(10),"AAAAAAAAAA");
+        Assert.assertEquals(new Kmer(11).toString(11),"AAAAAAAAAAA");
+    }
+
+    @Test
+    public void testConstructionFromStringsAndBytes()
+    {
+        testConstruction("ACGTACGTACGT");
+        testConstruction("ACGTACGTACGTC");
+        testConstruction("ACGTACGTACGTCC");
+        testConstruction("ACGTACGTACGTCCC");
+        testConstruction("ACGTACGTACGTCCCC");
+    }
+
+    private void testConstruction( final String str )
+    {
+        Assert.assertEquals(str,StringKmerizer.toKmer(str).toString(str.length()));
+        Assert.assertEquals(str,BytesKmerizer.toKmer(str.getBytes()).toString(str.length()));
+    }
+
+    @Test
+    public void testSuccessor()
```

Comment:
if you use the dataproviders you can use the same data provider for many tests if you want to, so you could reuse the same data provider for this as test construction